gnd/src/main/java/com/google/android/gnd/persistence/local/room/FormEntity.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.persistence.local.room;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.room.ColumnInfo;
+import androidx.room.Entity;
+import androidx.room.ForeignKey;
+import androidx.room.Index;
+import androidx.room.PrimaryKey;
+import com.google.android.gnd.model.form.Element;
+import com.google.android.gnd.model.form.Form;
+import com.google.auto.value.AutoValue;
+import com.google.auto.value.AutoValue.CopyAnnotations;
+import com.google.common.collect.ImmutableList;
+
+@AutoValue
+@Entity(
+    tableName = "form",
+    foreignKeys =
+        @ForeignKey(
+            entity = LayerEntity.class,
+            parentColumns = "id",
+            childColumns = "layer_id",
+            onDelete = ForeignKey.CASCADE),
+    indices = {@Index("layer_id")})
+public abstract class FormEntity {
+
+  @CopyAnnotations
+  @NonNull
+  @PrimaryKey
+  @ColumnInfo(name = "id")
+  public abstract String getId();
+
+  @CopyAnnotations
+  @Nullable
+  @ColumnInfo(name = "title")
+  public abstract String getTitle();
+
+  @CopyAnnotations
+  @NonNull
+  @ColumnInfo(name = "layer_id")
+  public abstract String getLayerId();
+
+  public static FormEntity fromForm(String layerId, Form form) {
+    return FormEntity.builder()
+        .setId(form.getId())
+        .setLayerId(layerId)
+        .setTitle(form.getTitle())
+        .build();
+  }
+
+  public static Form toForm(FormData formData) {
+    FormEntity formEntity = formData.formEntity;
+    Form.Builder formBuilder =
+        Form.newBuilder().setId(formEntity.getId()).setTitle(formEntity.getTitle());
+
+    ImmutableList.Builder<Element> listBuilder = ImmutableList.builder();
+    for (FieldData fieldData : formData.fields) {
+      for (ElementEntity elementEntity : formData.elementEntities) {
+        if (fieldData.fieldEntity.getId().equals(elementEntity.getFieldId())) {
+          listBuilder.add(ElementEntity.toElement(elementEntity, fieldData));
+        }
+      }
+    }
+
+    return formBuilder.setElements(listBuilder.build()).build();
+  }
+
+  public static FormEntity create(String id, String title, String layerId) {
+    return builder().setId(id).setTitle(title).setLayerId(layerId).build();
+  }
+
+  public static Builder builder() {
+    return new AutoValue_FormEntity.Builder();
+  }
+
+  @AutoValue.Builder
```

Comment:
Random thought: Should we just get rid of the Builders at some point and use the `create()` method? Android Studio shows us param names inline, so from a readability perspective there's little difference, and we need the create() method for Room. I'm undecided, but it might reduce some boilerplate.

Reply:
Sure, let's do this discussion separately in a refactor issue.